internal/integration/provision/upgrade.go

```diff
@@ -75,7 +75,7 @@ const (
 
 	previousK8sVersion = "1.20.1"
 	stableK8sVersion   = "1.20.4"
-	currentK8sVersion  = "1.20.4"
+	currentK8sVersion  = "1.20.5"
```

Comment:
they relate to the variables above, which describe Talos versions, so that is version of kubernetes matching that Talos version.

usually 

previous = previous stable (0.8.x)
stable = current stable (0.9.x)
current = master (0.10.x)

at the moment 0.9.x doesn't exist yet, so it's some 0.9.x-alpha